### PR TITLE
filter out empty string

### PR DESCRIPTION
**Pull request by james-ray**
* [x] Updated all relevant documentation in docs
* [x] Updated all code comments where relevant
* [x] Wrote tests
* [x] Updated CHANGELOG_PENDING.md

fix #2322

### CODE REVIEW

File: node/strings.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package node
+
+import "strings"
+
+//move from string.go. filter out empty string too.
+func SplitAndTrim(s, sep, cutset string) []string {
```
Comment (melekes) [inline node/strings.go]: let's make it private

File: node/strings.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package node
+
+import "strings"
+
+//move from string.go. filter out empty string too.
```
Comment (melekes) [inline node/strings.go]: The proper godoc should be in form of `splitAndTrim ...`

File: node/strings.go [hunk 3]
```diff
@@ -2,7 +2,8 @@ package node
 
 import "strings"
 
-//move from string.go. filter out empty string too.
+// SplitAndTrimEmpty is copied from common/string.go
```
Comment (ValarDragon) [inline node/strings.go]: Whats the point of saying that it is copied from `common/string.go`, we can just delete it from there as its only used here.

**Comment by ValarDragon** [comment]
Please don't keep amending the commits, we can squash it when merging. Constantly amending the commit makes it much harder to keep track of what the change between commits is :)

**Comment by james-ray** [comment]
Sorry for the commits loss thing.

File: libs/common/string.go [hunk 4]
```diff
@@ -58,6 +58,23 @@ func SplitAndTrim(s, sep, cutset string) []string {
 	return spl
 }
 
+// SplitAndTrimEmpty only returns non-empty strings
+func SplitAndTrimEmpty(s, sep, cutset string) []string {
```
Comment (melekes) [inline libs/common/string.go]: Please move it to `node.go`

File: node/node.go [hunk 5]
```diff
@@ -745,3 +746,21 @@ func saveGenesisDoc(db dbm.DB, genDoc *types.GenesisDoc) {
 	}
 	db.SetSync(genesisDocKey, bytes)
 }
+
+
+// SplitAndTrimEmpty returns only non-empty strings
+func SplitAndTrimEmpty(s, sep, cutset string) []string {
```
[the remark this answers is not in the document]
Comment (james-ray) [reply node/node.go]: oh,really,can I delete it? I'll check if anywhere used it. I will leave a changelog.

File: CHANGELOG.md [hunk 6]
```diff
@@ -38,6 +38,7 @@ IMPROVEMENTS:
     - tweak params
     - only process one block at a time to avoid starving
 - [common] bit array functions which take in another parameter are now thread safe
+- [common] SplitAndTrim was deleted
```
Comment (ValarDragon) [inline CHANGELOG.md]: Its kind of confusing, but we put changelog updates for the next release in `CHANGELOG_PENDING.md`, could you move this there, and put it in the breaking changes section?

**Comment by ValarDragon** [comment]
Looks good to me! Thank you for the PR! Just needs the changelog update moved to `CHANGELOG_PENDING.md`

**Comment by melekes** [comment]
Thanks for your contribution 🖖

**Pull request merged**